src/qt/bitcoingui.cpp

```diff
@@ -370,10 +370,28 @@ void BitcoinGUI::createActions()
             for (std::string path : m_wallet_controller->getWalletsAvailableToOpen()) {
                 QString name = path.empty() ? QString("["+tr("default wallet")+"]") : QString::fromStdString(path);
                 QAction* action = m_open_wallet_action->menu()->addAction(name);
-                connect(action, &QAction::triggered, [this, path] {
+                connect(action, &QAction::triggered, [this, name, path] {
                     OpenWalletActivity* activity = m_wallet_controller->openWallet(path);
+
+                    QProgressDialog* dialog = new QProgressDialog(this);
+                    dialog->setLabelText(tr("Opening Wallet <b>%1</b>...").arg(name.toHtmlEscaped()));
+                    dialog->setRange(0, 0);
+                    dialog->setCancelButton(nullptr);
```

Comment:
Shouldn't this cancel the rescan?

Reply:
Ok, but then I probably shouldn't be able to close the dialog either? At least not without a warning that rescan will continue.